src/sdl2/timer.rs

```diff
@@ -117,16 +118,14 @@ impl<'b, 'a> Drop for Timer<'b, 'a> {
     }
 }
 
-extern "C" fn c_timer_callback(_interval: u32, param: *mut c_void) -> u32 {
-    // FIXME: This is UB if the callback panics! (But will realistically
-    // crash on stack underflow.)
-    //
-    // I tried using `std::panic::catch_unwind()` here and it compiled but
-    // would not catch. Maybe wait for `c_unwind` to stabilize? Then the behavior
-    // will automatically abort the process when panicking over an `extern "C"`
-    // function.
```

Comment:
I don't know why that did not work, it should work.

Since Rust 1.81, unwinding in an `extern "C"` function will abort the process, I added the `catch_unwind` to not have this make undefined behavior on "older" ([1.81 is almost 2 weeks old at the time of writing this comment](https://releases.rs/docs/1.81.0/)) versions and have the same behavior independent of the Rust version.